apis/v1alpha1/validation/validation.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"net"
+	"strings"
+
+	gatewayv1a1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateGateway validates Gateway objects.
+func ValidateGateway(gw *gatewayv1a1.Gateway) field.ErrorList {
+	return ValidateGatewaySpec(&gw.Spec, field.NewPath("spec"))
+}
+
+// ValidateGatewaySpec validates whether required fields of spec are set according to the
+// Gateway API specification.
+func ValidateGatewaySpec(spec *gatewayv1a1.GatewaySpec, path *field.Path) field.ErrorList {
```

Comment:
Does this really need to be exported? Better for everyone to just use `ValidateGateway` IMHO.

Reply:
ack